nautobot/ipam/filters.py

```diff
@@ -429,6 +430,25 @@ def filter_virtual_machine(self, queryset, name, value):
         return queryset.filter(vm_interfaces__in=interface_ids)
 
 
+class IPAddressToInterfaceFilterSet(NautobotFilterSet):
+    interface = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=Interface.objects.all(),
+        label="Interface (name or ID)",
+    )
+    ip_address = django_filters.ModelMultipleChoiceFilter(
+        queryset=IPAddress.objects.all(),
+        label="IP Address (ID)",
+    )
+    vm_interface = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=VMInterface.objects.all(),
+        label="VM Interface (name or ID)",
+    )
```

Comment:
Should we continue to propagate the pattern of using `NaturalKeyOrPKMultipleChoiceFilter` for non-unique `name` values, given that we've already received feedback like #4207 and #4279? Or should these just use `ModelMultipleChoiceFilter` for now?

Reply:
Is it better to be inconsistent or consistently wrong? (if this filter is indeed wrong)

Other than the name of the filter containing "natural key" I wouldn't assume that if I filter on `?interface=Ethernet1` that I should expect a unique instance returned. We could go back to `interface` and `interface_id` filters but we just moved away from that pattern everywhere else in Nautobot so I don't think the decision to start walking back those changes should start here.

Reply:
IIRC the issue (you might want to test and confirm this) is that the current filter UI prefers "natural keys" over PKs, so if you select "Device 1 -> Interface Ethernet1" it'll give you `?interface=Ethernet1` (which is incorrect here) instead of `?interface=<UUID>`. But your point is well taken.

Reply:
Yeah I just verified that behavior on demo. Since we are presenting a select widget in the filter UI with specific instances as choices I would expect it to use the pk but it's using the "natural key"

Reply:
I don't feel strongly either way, if you want to make the call or we can bring it up for discussion tomorrow

Reply:
Per team discussion we'll stay consistently wrong here for now, plan to fix this more generally in addressing #4207/#4279.